sigilrecorder.lic

```diff
@@ -0,0 +1,297 @@
+			  return nil
+			end 
+
+			#main 'perc sigil' DRC.bput and case statement. When sigils are detected they're added to an array
+			case line = DRC.bput('perceive sigil', @sigil_list, 'Roundtime', /^You recall.*/, /^Having recently been searched/, /You are too distracted/, /Something about|This is not an appropriate place for that/)
+			when /^You recall.*/
+			  DRC.message('Room searched too recently') if @debug
+			  return nil
+			when /You are too distracted/
+			  prune_walk_list(Room.current.id, @data)
+			  return nil
+			when /Something about|This is not an appropriate place for that/
+			  prune_walk_list(Room.current.id, @data)
+			  return nil
+			when /^Having recently been searched/
+			  return nil
+			when *@sigil_list
+			  room_sigils.push(line)
+			  DRC.message("Found: #{room_sigils}") if @debug
+			end
+		end #while
+
+		#return the array of all sigils discovered in the room (should be 2 unless something went wrong)
+		return room_sigils
+	end
+	
+	#generate the list of all rooms that contain the current room's Room.current.image data
+	def region_room_list
+		data = Map.list.select do |room|
+		  room.image == Room.current.image
+		end
+
+		#generate the array
+		roomlist = Map.list.select { |room| room.image == Room.current.image }.map(&:id)
+
+		#return the full list of rooms
+		return roomlist
+	end #region_room_list
+	
+	#writes a roomid to the nogo list which is subtracted from the walk_list when the script is run in the future
+	def prune_walk_list(banroom, file_data)
+		#initialize a new array if nogo is empty
+		file_data.SigilInfo['nogo'] ||= []
+
+		#push the roomid to the array and remove any duplicates
+		file_data.SigilInfo['nogo'].push(banroom) unless file_data.SigilInfo['nogo'].include?(banroom)
+
+		#alert the player which room has been added to the nogo list
+		DRC.message("Unsuitable room for sigil harvesting. Adding Room #{banroom} to no-go list.") if @debug
+		File.open("scripts/data/base-sigils.yaml", "w") { |file| file.write(file_data.to_yaml) } #write roomid to file
+		Flags.reset('sigil-danger-room')
+		pause 0.2
+	end #prune_walk_list
+	
+	#reads base hunting data and combines all rooms with rooms from nogo into a single array
+	def generate_nogo_list(room_list)
+		DRC.message("Pruning walk list to avoid combat. Current list size: #{room_list.length}") if @debug
+		ban_list = @hunting_zones
+		ban_list += @data.SigilInfo['nogo'] unless @data.SigilInfo['nogo'] == nil
+
+		#subtract array of potentially dangerous rooms from the room list that was passed in
+		room_list -= ban_list
+		DRC.message("Walk list has been pruned. New list size: #{room_list.length}") if @debug
+		pause 1
+		return room_list #pruned, safer roomlist is returned
+	end #generate_nogo_list
+
+	#remove any rooms in the nogo list from recorded sigil locations before the script exits
+	def prune_final_list
+		ban_list = @hunting_zones
+		ban_list += @data.SigilInfo['nogo']
```

Comment:
Line 240 conditionally adds this only if it's not nil. Should this do the same thing?

Also, another low-code way to remove nil values from an array is to use  `compact` method:
```ruby
some_list = [ listA, listB, listC ].flatten.compact
```
Using `compact`, any nil elements are removed

Reply:
Collapsed to a single line: `ban_list = [@hunting_zones, @data.SigilInfo['nogo']].flatten.compact`